source/adios2/helper/adiosXML.cpp

```diff
@@ -11,473 +11,162 @@
-        // if can't find closing it's open until the end
-        if (closingQuotePosition == content.npos)
-        {
-            currentPosition == content.npos;
-            break;
-        }
-
-        currentPosition = closingQuotePosition + 1;
-
-        if (closingQuotePosition < foundPosition)
-        {
             continue;
         }
-        else
-        {
-            // if this point is reached it means it's a value inside " " or ' ',
-            // iterate
-            foundPosition = content.find(input, currentPosition);
-            currentPosition = foundPosition;
-        }
-    }
-
-    return foundPosition;
-}
 
-Params GetTagAttributesXML(const std::string &tagHeader)
-{
-    auto lf_GetQuotedValue = [](const char quote,
-                                const std::string::size_type &quotePosition,
-                                std::string &currentTag) -> std::string {
-
-        currentTag = currentTag.substr(quotePosition + 1);
-        auto nextQuotePosition = currentTag.find(quote);
-
-        if (nextQuotePosition == currentTag.npos)
+        pugi::xml_attribute attrValue = paramNode.attribute("value");
+        if (!attrValue)
         {
-            throw std::invalid_argument(
-                "ERROR: Invalid attribute in..." + currentTag +
-                "...check XML file, in call to ADIOS constructor\n");
-        }
-
-        const std::string value(currentTag.substr(0, nextQuotePosition));
-        currentTag = currentTag.substr(nextQuotePosition + 1);
-        return value;
-    };
-
-    auto lf_GetAttributes = [&](const std::string &tag) -> Params {
-        Params attributes;
-        std::string currentTag(tag.substr(tag.find_first_of(" \t\n")));
-        std::string::size_type currentPosition(0);
-
-        while (currentTag.find('=', currentPosition) !=
-               currentTag.npos) // equalPosition
-        {
-            currentTag = currentTag.substr(
-                currentTag.find_first_not_of(" \t\n", currentPosition));
-            auto equalPosition = currentTag.find('=');
-            if (currentTag.size() <= equalPosition + 1)
-            {
-                throw std::invalid_argument(
-                    "ERROR: tag " + tag +
-                    " is incomplete, check XML config file, "
-                    "in call to ADIOS constructor\n");
-            }
-
-            std::string key(currentTag.substr(0, equalPosition));
-            key.erase(key.find_last_not_of(" \t\n") + 1);
-
-            std::string value;
-
-            auto quotePosition =
-                currentTag.find_first_not_of(" \t\n", equalPosition + 1);
-
-            const char quote = currentTag.at(quotePosition);
-            if (quote == '\'' || quote == '"')
+            if (debugMode)
             {
-                value = lf_GetQuotedValue(quote, quotePosition, currentTag);
+                throw std::runtime_error(
+                    "No \"value\" attribute found on <parameter> element.");
```

Comment:
throw std::runtime_error("ERROR: in XML config file, no \"value\" attribute found on <parameter> element" + tagHeader + " in call to ADIOS constructor\n");